riscv/src/small_field/runtime.rs

```diff
@@ -0,0 +1,354 @@
+                    link ~> (tmp2_h, tmp2_l) = regs.mload(0, YL, STEP)
+                    link ~> (tmp3_h, tmp3_l) = add_sub.add(tmp2_h, tmp2_l, ZH, ZL)
+                    link ~> (tmp4_l, tmp4_h) = shift.shr(tmp1_l, tmp1_h, tmp3_l)
+                    link ~> regs.mstore(0, WL, STEP + 3, tmp4_h, tmp4_l);
+"#,
+            ],
+            0,
+            ["shl 0, 0, 0, 0, 0;"],
+        );
+
+        r.add_submachine::<&str, _, _>(
+            "std::machines::range::Bit2",
+            None,
+            "bit2",
+            vec![],
+            [],
+            0,
+            [],
+        );
+
+        r.add_submachine::<&str, _, _>(
+            "std::machines::range::Bit6",
+            None,
+            "bit6",
+            vec![],
+            [],
+            0,
+            [],
+        );
+
+        r.add_submachine::<&str, _, _>(
+            "std::machines::range::Bit7",
+            None,
+            "bit7",
+            vec![],
+            [],
+            0,
+            [],
+        );
+
+        r.add_submachine::<&str, _, _>(
+            "std::machines::range::Byte",
+            None,
+            "byte",
+            vec![],
+            [],
+            0,
+            [],
+        );
+
+        r.add_submachine::<&str, _, _>(
+            "std::machines::range::Bit12",
+            None,
+            "bit12",
+            vec![],
+            [],
+            0,
+            [],
+        );
+
+        r.add_submachine::<&str, _, _>(
+            "std::machines::range::Byte2",
+            None,
+            "byte2",
+            vec![],
+            [],
+            0,
+            [],
+        );
+
+        r.add_submachine::<&str, _, _>(
+            "std::machines::binary::ByteBinary",
+            None,
+            "byte_binary",
+            vec![],
+            [],
+            0,
+            [],
+        );
+
+        r.add_submachine::<&str, _, _>(
+            "std::machines::small_field::shift::ByteShift",
+            None,
+            "byte_shift",
+            vec![],
+            [],
+            0,
+            [],
+        );
+
+        r.add_submachine::<&str, _, _>(
+            "std::machines::split::ByteCompare",
+            None,
+            "byte_compare",
+            vec![],
+            [],
+            0,
+            [],
+        );
+
+        // Base syscalls
+        r.add_syscall(
+            // TODO this is a quite inefficient way of getting prover inputs.
+            // We need to be able to access the register memory within PIL functions.
+            Syscall::Input,
+            [
+                "query_arg_1_h, query_arg_1_l <== get_reg(10);",
+                "query_arg_2_h, query_arg_2_l <== get_reg(11);",
+                // TODO fix for BabyBear
+                "set_reg 10, 0, ${ std::prelude::Query::Input(std::convert::int(std::prover::eval(query_arg_1_l)), std::convert::int(std::prover::eval(query_arg_2_l))) };",
```

Comment:
@pacheco I think these are fine for this PR to keep going but we should make a plan soon.

Reply:
yeah, left a couple of comments but the PR lgtm so we can move on and iterate